### PR TITLE
+ support & tests for iOS 10+, Safari 10+, macOS Sierra

### PR DESCRIPTION
**Pull request by jacobpierce**

### CODE REVIEW

**Comment by jacobpierce** [comment]
commit message typo -- commit `2d14b1a` resolves issue #5 not #1

File: system.go [hunk 1]
```diff
@@ -251,7 +251,9 @@ func getiOSVersion(uaPlatformGroup string) int {
 	if len(s) == 0 {
```
Comment (ryanslade) [inline system.go]: This check should change since we're slicing below from len(s)-2, so if that ends up being negative it'll panic.

**Comment by ryanslade** [comment]
Review complete

File: system.go (hunk 1 above)
Comment (jacobpierce) [inline system.go]: Line 250 defines `s` as a match to the RegEx `(cpu|iphone) os \\d+`. Line 251 handles when there is no match, and exits as 0 (unknown). Any positive regex match would be 8+ char length and end in one or more digits so should not be able to panic, correct?

Comment (ryanslade) [inline system.go]: Cool, you're right.

File: system.go [hunk 2]
```diff
@@ -251,7 +251,9 @@ func getiOSVersion(uaPlatformGroup string) int {
...
 		return 0
 	}
-	i := strToInt(string(s[len(s)-1]))
+
+	// catpure and trim the last 2 characters; convert to int
+	i := strToInt(strings.TrimSpace(string(s[len(s)-2:])))
```
Comment (ryanslade) [inline system.go]: One minor thing is that you don't need the string() conversion here since s is already a string.

Comment (jacobpierce) [inline system.go]: :+1: updated

**Pull request merged**